dd-sdk-android/src/test/kotlin/com/datadog/android/domain/LogSerializerTest.kt

```diff
@@ -0,0 +1,210 @@
+package com.datadog.android.domain
+
+import com.datadog.android.log.assertj.JsonObjectAssert.Companion.assertThat
+import com.datadog.android.log.forge.Configurator
+import com.datadog.android.log.internal.domain.Log
+import com.datadog.android.log.internal.domain.LogSerializer
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.google.gson.JsonPrimitive
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.util.Date
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.STRICT_STUBS)
+@ForgeConfiguration(Configurator::class)
+internal class LogSerializerTest {
+
+    lateinit var underTest: LogSerializer
+
+    @BeforeEach
+    fun `set up`() {
+        underTest = LogSerializer()
+    }
+
+    @Test
+    fun `serializes full log as json`(@Forgery fakeLog: Log) {
+        val serialized = underTest.serialize(fakeLog)
+        assertLogMatches(serialized, fakeLog)
+    }
+
+    @Test
+    fun `ignores reserved attributes`(@Forgery fakeLog: Log, forge: Forge) {
+        // given
+        val logWithoutAttributes = fakeLog.copy(attributes = emptyMap())
+        val attributes = forge.aMap {
+            anElementFrom(*LogSerializer.reservedAttributes) to forge.anAsciiString()
+        }.toMap()
+        val logWithReservedAttributes = fakeLog.copy(attributes = attributes)
+
+        // when
+        val serialized = underTest.serialize(logWithReservedAttributes)
+
+        // then
+        assertLogMatches(serialized, logWithoutAttributes)
+    }
+
+    @Test
+    fun `ignores reserved tags keys`(@Forgery fakeLog: Log, forge: Forge) {
+        // given
+        val logWithoutTags = fakeLog.copy(tags = emptyList())
+        val key = forge.anElementFrom("host", "device", "source", "service")
+        val value = forge.aNumericalString()
+        val reservedTag = "$key:$value"
+        val logWithReservedTags = fakeLog.copy(tags = listOf(reservedTag))
+
+        // when
+        val serialized = underTest.serialize(logWithReservedTags)
+
+        // then
+        assertLogMatches(serialized, logWithoutTags)
+    }
+
+    @Test
+    fun `asserts a log with no network info available`(@Forgery fakeLog: Log, forge: Forge) {
```

Comment:
This is a test class, you don't  need to add *asserts*  to the test method names. Best practice is to remove "test", "should"… from method name, and simply describe the behavior we wanna test. eg: "ignores reserved tags keys". For this method you can simply say "serializes log with no network info".